File: src/main/java/org/ffpy/plugin/coding/util/CopyPasteUtils.java
package org.ffpy.plugin.coding.util;

import com.intellij.designer.clipboard.SimpleTransferable;
import com.intellij.openapi.ide.CopyPasteManager;

import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;
import java.util.Optional;

public class CopyPasteUtils {

    public static void putString(String content) {
        CopyPasteManager.getInstance().setContents(new SimpleTransferable(content, DataFlavor.stringFlavor));
    }

    public static Optional<String> getString() {
        return Optional.ofNullable(CopyPasteManager.getInstance().getContents())
                .filter(contents -> contents.isDataFlavorSupported(DataFlavor.stringFlavor))
                .map(contents -> {
                    try {
                        return (String) contents.getTransferData(DataFlavor.stringFlavor);
                    } catch (UnsupportedFlavorException | IOException e) {
                        return null;
                    }
                });
    }
}

File: src/main/kotlin/org/ffpy/plugin/coding/util/CopyPasteUtils.kt
package org.ffpy.plugin.coding.util

import com.intellij.designer.clipboard.SimpleTransferable
import com.intellij.openapi.ide.CopyPasteManager
import java.awt.datatransfer.DataFlavor

object CopyPasteUtils {

    fun putString(content: String?) {
        CopyPasteManager.getInstance().setContents(SimpleTransferable(content, DataFlavor.stringFlavor))
    }

    fun getString(): String? {
        return CopyPasteManager.getInstance().contents?.let {
            if (it.isDataFlavorSupported(DataFlavor.stringFlavor)) {
                return try {
                    it.getTransferData(DataFlavor.stringFlavor) as String
                } catch (e: Exception) {
                    null
                }
            } else {
                null
            }
        }
    }
}